go/mysql/schema.go

```diff
@@ -32,11 +32,11 @@ import (
 
 const (
 	// BaseShowPrimary is the base query for fetching primary key info.
```

Comment:
We are breaking compatibility with MySQL here, right? It might be a breaking change for consumers of the schema's PKColumns (we use the same query to compute that), since some tables which advertised a PK will no longer to so. 

 At least VTExplain and schema tracker uses this, not sure if any other non-vreplication modules do.

Reply:
And keep in mind https://github.com/vitessio/vitess/pull/9794 is only in v14.0-SNAPSHOT, and THAT is what actually changed the behavior here (that PR is what _unintentionally_ addressed https://github.com/vitessio/vitess/issues/7120). My PR takes it back to only providing the actual defined PKs.

Reply:
Found some context on the selection of unique key as the primary key: https://bugs.mysql.com/bug.php?id=11228

As to which unique key is selected is still ambiguous:

A comment in this issue suggests it is the first unique key in the create ddl:
"The intention with the original code was to show that MySQL treats the first given unique key as a primary key. Clients can use the marked primary key as a real primary key to validate row changes in case of conflicting updates.  The ODBC driver (and other drivers) may also use this fact to optimize/check updates."

BUT, in the mysql docs it just says "A UNIQUE index may be displayed as PRI if it cannot contain NULL values and there is no PRIMARY KEY in the table".

My guess is that they needed to select a key for a partial binlog_row_image, since the partial row, in addition to the changed columns, needs to contain the columns to uniquely identify a row.

Anyway, we don't need to worry about this for Vitess, just noting it here for completeness.

Reply:
IMHO, and for predictability, we should always rely on our function rather than the PKE MySQL suggests (which is news to me, I wasn't aware it can do that).

- Firstly, because it looks like MySQL isn't deterministic about the key it chooses (consider we can have two near identical tables with a changed ordering of keys - they may produce different PKEs).
- Second, because our query can at least heuristiclaly optimize for the better (ie smaller) UK